From: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
Date: Fri, 8 Nov 2019 23:33:07 -0500
Subject: Avoid weird Preprocessor.py misbehavior

Sometimes, when util/__pycache__ is not populated,
util/Preprocessor.py fails to produce any output.

I have not been able to diagnose why this happens, but this will
hopefully work around it.
---
 ui/content/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ui/content/Makefile b/ui/content/Makefile
index 0cdbd60..4b2ab3c 100644
--- a/ui/content/Makefile
+++ b/ui/content/Makefile
@@ -139,7 +139,7 @@ $(GENDIR)/%.js: %.js
 	$(DEPTH)/util/prepPostbox -c $(TARGET_TOOL) $< $@
 
 $(GENDIR)/%.xul: %.xul
-	$(PYTHON) $(DEPTH)/util/Preprocessor.py -I $(INCLUDE) -o $@ $<
+	if [ -s $< ]; then while [ ! -s $@ ]; do $(PYTHON) $(DEPTH)/util/Preprocessor.py -I $(INCLUDE) -o $@ $<; done; else touch $@; fi
 
 $(GENDIR)/%.htm: %.htm
 	$(DEPTH)/util/install -m 644 $(DIST)/chrome/content/ui $<
